test/e2e/framework/framework.go

```diff
@@ -292,8 +292,7 @@ func (f *Framework) WaitForGameServerState(t *testing.T, gs *agonesv1.GameServer
 // Each Allocated GameServer gets deleted allocDuration after it was Allocated.
 // GameServers will continue to be Allocated until a message is passed to the done channel.
 func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
-	// nolint:staticcheck
-	err := wait.PollImmediateUntil(period, func() (bool, error) {
+	err := wait.PollUntilContextCancel(ctx, period, true, func(ctx context.Context) (bool, error) {
```

Comment:
Looks like this doesn't work because PollUntilContextCancel doesn't return ErrWaitTimeout error so error was thrown in line 315. Please revert the change for this line.

Reply:
It will be annotated with // nolint:staticcheck?

Reply:
Yes just revert the change you made for this line in this PR

Reply:
Below code will work I guess,
```
func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
	err := wait.PollUntilContextCancel(ctx, period, true, func(ctx context.Context) (bool, error) {
		gsa := GetAllocation(flt)
		gsa, err := f.AgonesClient.AllocationV1().GameServerAllocations(flt.Namespace).Create(ctx, gsa, metav1.CreateOptions{})
		if err != nil || gsa.Status.State != allocationv1.GameServerAllocationAllocated {
			// Ignore error. Could be that the buffer was empty, will try again next cycle.
			return false, nil
		}

		// Deallocate after allocDuration.
		go func(gsa *allocationv1.GameServerAllocation) {
			time.Sleep(allocDuration)
			err := f.AgonesClient.AgonesV1().GameServers(gsa.Namespace).Delete(ctx, gsa.Status.GameServerName, metav1.DeleteOptions{})
			require.NoError(t, err)
		}(gsa)

		return false, nil
	})
	if err != nil {
		require.NoError(t, err)
	}
}
```

Reply:
will work on this in the upcoming PR